src/dpl/src/PlacementDRC.cpp

```diff
@@ -302,7 +302,9 @@ bool PlacementDRC::checkPadding(const Node* cell,
     for (GridY grid_y{y}; grid_y < cell_y_end; grid_y++) {
       const Pixel* pixel = grid_->gridPixel(grid_x, grid_y);
       if (pixel == nullptr) {  // at the core edge
-        if (!cell->isFixed()) {
+        const bool left_edge = x == 0;
+        const bool right_edge = cell_x_end == grid_->getRowSiteCount();
+        if (!cell->isFixed() && !left_edge && !right_edge) {
           // Padding violation with the core edge.
           return false;
         }
```

Comment:
When would `pixel` ever be null without being at the core edge? I remember spending time on this case when I first introduced the new padding handling in DPL, and I didn’t want to alter the existing results. That’s why I added this check to return false if there isn’t enough padding between the cell and the core edge. If that’s no longer necessary, then just change all of this to:
```
if (pixel == nullptr)
{
  continue;
}
```

Reply:
I could imagine a scenario where an instance has 1-site distance fro the core edge, and the padding is 2. If I understand correctly, the `pixel` would still be `nullptr`, and a potential error would occur, where we'd have a 1-site gap between the cell and the core edge. That's why I decide to ensure the instance is adjacent to the edge.

While writing it, I realized I should have added a test for this scenario, at least to confirm that this is what happens. I'll try it later.